datafusion/core/src/lib.rs

```diff
@@ -569,10 +569,10 @@ pub mod functions {
     pub use datafusion_functions::*;
 }
 
-/// re-export of [`datafusion_functions_array`] crate, if "array_expressions" feature is enabled
-pub mod functions_array {
-    #[cfg(feature = "array_expressions")]
-    pub use datafusion_functions_array::*;
+/// re-export of [`datafusion_functions_nested`] crate, if "nested_expressions" feature is enabled
```

Comment:
Maybe we can leave a `pub use functions_array` here to help migration

Something like

```rust
    #[deprecated(since = "41.0.0", note = "use datafusion-functions-nested instead")]
pub mod functions_array {
    #[cfg(feature = "nested_expressions")]
    pub use datafusion_functions_nested::*;
}
```

We could do something similar with the feature flags, but maybe that is too complicated to be worth it

Reply:
Good idea. Thanks. I'll try to do it for the crate and feature.